book/src/multithreading.md

```diff
@@ -0,0 +1,45 @@
+# Multithreading
+
+RIOT-rs supports multithreading on the Cortex-M, RISC-V, and Xtensa architectures.
+
+## Scheduling
+
+- **Fixed-priority, preemptive scheduling** policy with up to 32 supported priority levels.
+- **Thread priorities are dynamic** and can be changed at runtime.
+- The Runqueue is implemented as a circular linked list for each priority, requiring n × m × 8 bits for _n_ maximum threads and _m_ supported priorities. All operations on the runqueue are performed in constant time, except for the deletion of a thread that is not the head of the runqueue.
+- **Deep sleep when idle**: On single core, no idle threads are created. Instead, if the runqueue is empty, the processor enters deep sleep until a next thread is ready. The context of the previously running thread is only saved once the next thread is ready and the context switch occurs.
+- **Same priority threads are scheduled cooperatively.** The scheduler itself is tickless, therefore time-slicing isn't supported. [Timers](https://docs.rs/embassy-time/latest/embassy_time/struct.Timer.html) are still supported through the timer API from the integrated [Embassy] crate.
```

Comment:
This needs a previous "Policy: The highest priority runnable thread (or thread**s** in the multi-core case) is executed" or similar to make sense.

Reply:
Maybe drop the timers sentence?

Reply:
> This needs a previous "Policy: The highest priority runnable thread (or threads in the multi-core case) is executed" or similar to make sense.

I'd say that's implied by the _preemptive priority_ scheduling above.

Reply:
:+1: Done.